hera_cal/nucal.py

```diff
@@ -0,0 +1,473 @@
+        ]
+
+    def get_u_bounds(self, freqs):
+        """
+        Calculates the magnitude of the minimum and maximum u-modes values of the radial redundant group
+        given an array of frequency values
+
+        Parameters:
+        ----------
+        freqs: np.ndarray
+            Array of frequencies found in the data in units of Hz
+
+        Returns:
+        -------
+        ubounds: tuple
+            Tuple of the magnitude minimum and maximum u-modes sampled by this baseline group
+        """
+        umin = freqs.min() / SPEED_OF_LIGHT * np.min(self.baseline_lengths)
+        umax = freqs.max() / SPEED_OF_LIGHT * np.max(self.baseline_lengths)
+        return (umin, umax)
+
+    def filter_group(
+        self,
+        bls=None,
+        ex_bls=None,
+        ants=None,
+        ex_ants=None,
+        pols=None,
+        ex_pols=None,
+        min_bl_cut=None,
+        max_bl_cut=None,
+    ):
+        """
+        Filter radially redundant group to include/exclude the specified bls, antennas. and polarizations.
+        Arguments are evaluated, in order of increasing precedence: (pols, ex_pols, bls, ex_bls, ants, ex_ants,
+        min_bl_cut, max_bl_cut).
+
+        Parameters:
+        ----------
+        bls : list of tuples, default=None
+            baselines to include. Baselines of the form (i,j,pol) include that specific
+            visibility.  Baselines of the form (i,j) are broadcast across all polarizations present in reds.
+        ex_bls : list of tuples, default=None
+            same as bls, but excludes baselines.
+        ants : list of tuples, default=None
+            antennas to include. Only baselines where both antenna indices are in ants
+            are included.  Antennas of the form (i,pol) include that antenna/pol. Antennas of the form i are
+            broadcast across all polarizations present in reds.
+        ex_ants : list of tuples, default=None
+            same as ants, but excludes antennas.
+        pols : list of strings
+            polarizations to include in reds. e.g. ['nn','ee','ne','en']
+        ex_pols : list of strings
+            same as pols, but excludes polarizations.
+        min_bl_cut:
+            Cut baselines in the radially redundant group with lengths less than min_bl_cut
+        max_bl_cut:
+            Cut baselines in the radially redundant group with lengths less than min_bl_cut
+        """
+        _baselines = redcal.filter_reds(
+            [self._baselines],
+            bls=bls,
+            ex_bls=ex_bls,
+            ants=ants,
+            ex_ants=ex_ants,
+            pols=pols,
+            ex_pols=ex_pols,
+        )
+        if len(_baselines) == 0:
+            self._baselines = []
+            self.baseline_lengths = []
+        else:
+            new_bls = []
+            new_bls_lengths = []
+            for bls in _baselines[0]:
+                index = self._baselines.index(bls)
```

Comment:
doing all this index searching might be slower than recomputing baseline lengths from antpos, which honestly you probably want to store internally in the __init__

Reply:
If lengths are stored via a dictionary, it'd be faster. That's another option.